### PR TITLE
Remove an unnecessary `dynamic`

### PR DESCRIPTION
**Pull request by natebosch**
The `dynamic` was used to satisfy the purpose that is more directly
expressed with `FutureOr`.

Replace the private `_Command` class with a typedef record with named fields.

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]
needs a rebase, but yes!

**Comment by natebosch** [comment]
I expanded the scope a bit from the original review. @jakemac53 WDYT?

File: pkgs/test_core/lib/src/runner/console.dart [hunk 1]
```diff
@@ -101,17 +102,3 @@ class Console {
     }
   }
 }
-
-/// An individual console command.
-class _Command {
```
Comment (kevmoo) [inline pkgs/test_core/lib/src/runner/console.dart:106]: Mmm...records!

**Review by kevmoo — APPROVED** [review]

File: pkgs/test_core/lib/src/runner/console.dart [hunk 2]
```diff
@@ -13,7 +13,8 @@ import '../util/io.dart';
 /// An interactive console for taking user commands.
 class Console {
   /// The registered commands.
-  final _commands = <String, _Command>{};
+  final _commands = <String,
```
Comment (jakemac53) [inline pkgs/test_core/lib/src/runner/console.dart:16]: Fwiw, I don't like having the record type jammed in here - I would prefer a typedef for the record type?

**Review by jakemac53 — APPROVED** [review]

File: pkgs/test_core/lib/src/runner/console.dart (hunk 2 above)
Comment (natebosch) [reply pkgs/test_core/lib/src/runner/console.dart]: Added a typedef. I tried also with a trailing comma at the inline usage site and while I think it's a bit better than this version I don't think I like it as much as the typedef.

File: pkgs/test_core/lib/src/runner/console.dart [hunk 3]
```diff
@@ -102,16 +102,8 @@ class Console {
...
-
-  /// The single-line description of the command.
-  final String description;
-
-  /// The callback to run when the command is invoked.
-  final dynamic Function() body;
-
-  _Command(this.name, this.description, this.body);
-}
+typedef _Command = ({
```
Comment (kevmoo) [inline pkgs/test_core/lib/src/runner/console.dart:105]: That works, too!

**Pull request merged**